securedrop/source_user.py

```diff
@@ -118,6 +122,8 @@ def __init__(
         self._scrypt_p = scrypt_p
         self._backend = default_backend()
 
+    # Use @lru_cache to not recompute the same values over and over for the same user
+    @lru_cache
```

Comment:
This should help a lot with performance, by not having to recompute hashes on every authenticated request.